File: ISA-Backend/BloodBankAPI/Materials/DTOs/StaffRegistrationDTO.cs

using System.ComponentModel.DataAnnotations;

namespace BloodBankAPI.Materials.DTOs
{
    public class StaffRegistrationDTO
    {
        [Required]
        [DataType(DataType.EmailAddress)]
        public string Email { get; set; }
        [Required]
        public string Password { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string Surname { get; set; }
        [Required]            
        public string Gender { get; set; }
        [Required]
        public int CenterId { get; set; }

        public StaffRegistrationDTO() { }
    }
}

File: ISA-Frontend/src/app/model/staffRegistrationDTO.ts
export class StaffRegistrationDTO {
    email = ''
    password = ''
    userType = 'STAFF'
    name = ''
    surname = ''
    gender = ''
    centerId = 0
  
    public constructor(obj?: any) {
      if (obj) {
        this.email = obj.email
        this.password = obj.password
        this.name = obj.name
        this.surname = obj.surname
        this.gender = obj.gender
        this.centerId = obj.centerId
      }
    }
  }
  